File: 240123/빙빙 돌며 숫자 사각형 채우기/snail-number-square.py
n, m = map(int, input().split())
arr = [[0 for j in range(m)] for i in range(n)]

def in_range(x, y):
    return 0 <= x < n and 0 <= y < m

dx = [0, 1, 0, -1]
dy = [1, 0, -1, 0]

x, y = 0, 0
dir_num = 0

arr[x][y] = 1

for i in range(2, n * m + 1):
    nx = x + dx[dir_num]
    ny = y + dy[dir_num]

    if not in_range(nx, ny) or arr[nx][ny] != 0:
        dir_num = (dir_num + 1) % 4

    x = x + dx[dir_num]
    y = y + dy[dir_num]
    arr[x][y] = i

for i in range(n):
    for j in range(m):
        print(arr[i][j], end=' ')
    print()

File: 240911/빙빙 돌며 숫자 사각형 채우기/snail-number-square.js
const fs = require("fs");
let [n, m] = fs.readFileSync(0).toString().trim().split(" ").map(Number);

let arr = Array(n).fill(0).map(() => Array(m).fill(0));
const dx = [0, 1, 0, -1], dy = [1, 0, -1, 0];
let dirNum = 0;
let [x, y] = [0, 0];

arr[0][0] = 1;

function inRange(x, y) {
    return x >= 0 && x < n && y >= 0 && y < m;
}

for (let i = 2; i <= n * m; i++) {
    let nx = x + dx[dirNum], ny = y + dy[dirNum];

    if (!inRange(nx, ny) || arr[nx][ny] !== 0) {
        dirNum = (dirNum + 1) % 4;
    } 

    x = x + dx[dirNum], y = y + dy[dirNum];

    arr[x][y] = i;
    
}

for (let i = 0; i < n; i++) {
    console.log(arr[i].join(" "));
}